src/sagas/user.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import { put, call, takeLatest, all, select } from "redux-saga/effects";
+import type { Saga } from "redux-saga";
+import {
+  request,
+  requestError,
+  requestSuccess,
+  logoutSuccess,
+  type Login
+} from "../actions/user";
+import { USER } from "../constants/actionTypes";
+import { login } from "../api";
+
+export function* authorize(action: Login): Saga<void> {
```

Comment:
authorize means that you should pass authorization, fetched profile doesn't mean that you authorized.
If you want to create logic to authorize your app, you should fetch token at first, and only then fetch user.